code/API_definitions/geofencing.yaml

```diff
@@ -0,0 +1,785 @@
+    
+    If the geofencing-state of a device changes, the event subscriber will be notified back to the provided Notification-Url given by the subscription-request.
+    
+    Device geofencing API could be useful in scenarios such as:
+    
+    - Tracking devices for Presetting of Home-Settings 
+    - Tracking of logistics
+    
+    # Relevant terms and definitions
+    
+    * **Device**: A device refers to any physical entity that can connect to a network and participate in network communication.
+    
+    * **Area**: It specifies the geographical surface which a device is planned to enter or exit.
+    
+    # API Functionality
+    
+    The API exposes following capabilities:
+    
+    ## Device Geofencing subscription
+    These endpoints allow to manage event subscription on geofencing device location event. 
+    
+    The CAMARA subscription model is detailed in the CAMARA API design guideline document and follows CloudEvents specification.
+
+    It is mandatory in the subscription to provide the event `type` subscribed are several are managed in this API.
+    
+    Following event ``type`` are managed for this API:
+      - ``org.camaraproject.geofencing.v0.area-entered``  - Event triggered when the device enters the given area
+    
+      - ``org.camaraproject.geofencing.v0.area-left``     - Event triggered when the device leaves the given area
+    
+    Note: Additionally to these list, ``org.camaraproject.geofencing.v0.subscription-ends`` notification `type` is sent when the subscription ends. 
+    This notification did not requires dedicated subscription. 
+    It is used when the subscription expire time (required by the requester) has been reached or if the API server has to stop sending notification prematurely. 
+
+    ### Notification callback
+
+    This endpoint describes the event notification received on subscription listener side when the event occurred. As for subscription, detailed description of the event notification is provided in the CAMARA API design guideline document.
+
+    **WARNING**: This callback endpoint must be exposed on the listener side as `POST /{$request.body#/webhook/notificationUrl}`
+      will be posted by the device geofencing resource server to the webhook url provided by notification 
+      listener.
+    
+    # Further info and support
+    
+    (FAQs will be added in a later version of the documentation)    
+
+  termsOfService: http://swagger.io/terms/
+  contact:
+    email: project-email@sample.com
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+  version: 0.1.0-wip
+externalDocs:
+  description: Product documentation at Camara
+  url: https://github.com/camaraproject/
+
+servers:
+  - url: '{apiRoot}/{basePath}'
+    variables:
+      apiRoot:
+        default: http://localhost:9091
+        description: API root
+      basePath:
+        default: geofencing/v0
```

Comment:
I do not agree here, because "default" is a required property here. 

> Unlike [path parameters](https://swagger.io/docs/specification/describing-parameters/#path), server variables do not use a schema. Instead, they are assumed to be strings. Variables can have arbitrary values, or may be restricted to an enum. In any case, a default value is required, which will be used if the client does not supply a value.

Source https://spec.openapis.org/oas/v3.0.3#fixed-fields-4

Reply:
@maxl2287 @jlurien 
For location retrieval I've described as such:

```servers:
  - url: '{apiRoot}/location-retrieval/v0'
    variables:
      apiRoot:
        default: http://localhost:9091
        description: API root`
```

Looking for your guidance if it is correct.

Reply:
Reading the spec again, I agree that default is required if there is a variable, but also according to the example in the spec:

```
**basePath:
      # open meaning there is the opportunity to use special base paths as assigned by the provider**
```

I think we should not define a variable for basePath if we all agree that "/api-name/vX" is something that must NOT be changed by the provider. In that case, what @bigludo7 proposes is OK to me.